Automattic/jetpack
# My Jetpack: Add red bubble and warning Notice when paid plans are expired or expiring soon.

This PR adds red bubble functionality and displays a warning/error Notice in My Jetpack when any paid plans are expiring soon or expired.





Fixes https://github.com/Automattic/jetpack-roadmap/issues/1937

## Proposed changes:

*

### Other information:


## Jetpack product discussion



Project Thread: pbNhbs-bJN-p2

Design Thread: p1HpG7-umC-p2

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






- Apply this PR branch on a Jurassic Ninja site using the Beta plugin - In the "More options" section, select to assign a license for Jetpack VideoPress.
- Connect Jetpack (both site and user connection).
- Go to Store Admin, Look up the JN site, Make sure that Jetpack VideoPress subscription is attached to the site. Now change the expiration date to just a few days after today's date, so that it will be "expiring soon". (you may need to turn-off "auto-renew".. I'm not sure.)
- Go to My Jetpack and reload the page.
- Verify that the status of the VideoPress card is Expiring soon. (See image)
![Screen Shot on 2024-11-10 at 09-51-26](https://github.com/user-attachments/assets/ee5cfc10-ebc6-4ab7-a15d-055d1288a45f)
- Verify that there is a red bubble on the My Jetpack menu item.
![Screen Shot 2024-11-11 at 09 56 05](https://github.com/user-attachments/assets/c873dacc-642a-42f4-875f-ea26ea71d5bc)
- Verify that you see the following "Expiring soon" Notice:
![Screen Shot on 2024-11-10 at 09-55-31](https://github.com/user-attachments/assets/6f74bde4-f373-419c-b6f3-abf9d14ee20c)
- Down in the "My plans" section, verify your VideoPress plan is showing as "Expiring soon on..." with yellow color text, and showing a "Renew subscription" link. See screenshot:
![Screen Shot on 2024-12-02 at 11-32-22](https://github.com/user-attachments/assets/7e01bb11-d498-4f57-add4-12de1bc6bd13)
- In Store Admin, change the VideoPress expiring date to before todays date, so that it will be "Expired".
- Verify that the status of the VideoPress card is Expired. (See image)
![Screen Shot on 2024-11-10 at 14-30-49](https://github.com/user-attachments/assets/9141e52c-dd34-4057-8776-6ef64d82e7d8)
- Verify that there is a red bubble on the My Jetpack menu item.
- Verify that you see the following "Expired" Notice:
![Screen Shot on 2024-11-10 at 14-31-05](https://github.com/user-attachments/assets/3a45f25c-a8fc-4b95-826c-d020c476c2f3)
- Down in the "My plans" section, verify your VideoPress plan is showing as "Expired on..." with red color text, and showing a "Resume subscription" link. See screenshot:
![Screen Shot on 2024-12-02 at 11-38-02](https://github.com/user-attachments/assets/9f8070cc-87dd-4a61-b365-1fff82cd62c7)
- In Store Admin, remove the VideoPress subscription, and now add a Jetpack Security subscription. Run through the same steps as above where you'll set the expiration date to 'expiring soon", view the result, then set the expiration to 'expired', view the result, making sure the red-bubble and Notice are displaying properly.

"Expiring soon" Notice for Jetpack Security:

![Screen Shot on 2024-11-26 at 12-04-28](https://github.com/user-attachments/assets/29ad6713-f5ea-4933-a92f-7b4911e3c9a5)

"Expired Notice" for Jetpack Security:

![Screen Shot on 2024-11-26 at 11-03-02](https://github.com/user-attachments/assets/7187d2f1-31f6-4641-97d0-cf75cf4ba9f4)

- Next, disconnect Jetpack. Verify there are no code errors and that the statuses of all the product cards appear as expected.
- Now establish a site-only connection (no user connection). Verify there are no code errors and that the statuses of all the product cards appear as expected.
